spec/index.bs

```diff
@@ -1019,11 +1019,37 @@ conditions.
 <script type=idl>
 typedef (Uint32Array or DOMString) GPUShaderCode;
 
+typedef any GPUShaderSource;
+callback GPUShaderTransformFunction = GPUShaderCode (GPUShaderSource source);
+
 dictionary GPUShaderModuleDescriptor : GPUObjectDescriptorBase {
-    required GPUShaderCode code;
+    required GPUShaderSource source;
+    GPUShaderTransformFunction transform;
 };
 </script>
 
+{{GPUShaderModuleDescriptor}} has the following internal slots:
+<dl dfn-type=attribute dfn-for=GPUShaderModuleDescriptor>
+    : <dfn dfn>\[[code]]</dfn>, of type {{GPUShaderCode}}, readonly
```

Comment:
If you did keep this:

```suggestion
    : <dfn>\[[code]]</dfn>, of type {{GPUShaderCode}}, readonly
```

and necessary ref syntax changes below. (`{{GPUShaderModuleDescriptor/[[code]]}}`)